### ISSUE TITLE
Errors are not sorted by file

### ISSUE THREAD

**Issue by gvanrossum** [python/mypy#2144]
I just noticed that when you run mypy on a large number of files, the errors don't come out sorted by filename, even if you pass the filenames in sorted order on the command line. I presume this is some kind of side effect of processing files in an order determined by the dependency graph.

Maybe we should just sort them all by filename? Or sort them back in the order in which they were passed on the command line (or expanded from packages/directories)?

It seems the Errors.sort_messages() method in errors.py could easily sort messages by filename; sorting them by original order would require the original order to be passed in to that function (not rocket science, but perhaps requiring some small refactoring).

**Comment by gvanrossum** [comment]
(It's important not to attempt to sort the textual output, because it may contains note referring to other files in the import stack that led to a specific error.)

**Comment by elazarg** [comment]
+1

In general the output for `mypy *,py` is much easier to read than `mypy something.py` (though it's mainly because of the notes). I'd thought it should be the other way around.

**Comment by sebastian-philipp** [comment]
Yep, it makes updates of mypy reports hard to read:

https://gist.github.com/sebastian-philipp/25f70aae3b0d21b1a781c110a7ef8be4/revisions

**Comment by JukkaL** [comment]
Mypy now does error streaming, i.e. errors are reported as soon as they are generated. Sorting errors is incompatible with streaming.

**Comment by sebastian-philipp** [comment]
yep, it would need to store the errors and print everything at the end after the last file finishes.

**Comment by gvanrossum** [comment]
Well, it does still store them and print them later. They are printed after a file or a group of interconnected files (through import cycles) is processed. So they could be sorted within such a file (group).

Also, in daemon mode all errors come out at the very end (though we occasionally regret this) so it should be possible to sort them all.
